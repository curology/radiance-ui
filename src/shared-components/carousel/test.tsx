import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import './matchMedia.mock';
import { Carousel } from './index';

const cards = [<Carousel.Card key="card1">Card 1</Carousel.Card>];

describe('<Carousel />', () => {
  /**
   * TODO: Fix Emotion 11 CI snapshot serializer order issue
   */
  // eslint-disable-next-line jest/no-disabled-tests
  describe.skip('UI snapshots', () => {
    it('renders with props', () => {
      const { container } = render(
        <Carousel numCardsVisible={1}>{cards}</Carousel>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('renders with bottom right aligned arrows', () => {
      const { container } = render(
        <Carousel numCardsVisible={2} bottomRightAlignedArrows hideDots>
          {cards}
        </Carousel>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });
});
