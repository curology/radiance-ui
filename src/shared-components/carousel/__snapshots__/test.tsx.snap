// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> UI snapshots renders with bottom right aligned arrows 1`] = `
.emotion-0 {
  max-width: 702px;
}

.emotion-2 {
  overflow: hidden!important;
  padding-bottom: 1.5rem;
}

.emotion-2 .slick-slider {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 1.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 .slick-list {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  overflow: hidden;
}

.emotion-2 .slick-list:focus {
  outline: none;
}

.emotion-2 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.emotion-2 .slick-slider .slick-track,
.emotion-2 .slick-slider .slick-list {
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  -o-transform: translate3d(0, 0, 0);
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
}

.emotion-2 .slick-track {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 .slick-loading .slick-track {
  visibility: hidden;
}

.emotion-2 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.emotion-2 .slick-slide.slick-loading img {
  display: none;
}

.emotion-2 .slick-slide.dragging img {
  pointer-events: none;
}

.emotion-2 .slick-initialized .slick-slide {
  display: block;
}

.emotion-2 .slick-loading .slick-slide {
  visibility: hidden;
}

.emotion-2 .slick-vertical .slick-slide {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  border: 1px solid transparent;
}

.emotion-2 .slick-arrow.slick-hidden {
  display: none;
}

.emotion-2 .slick-slider {
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 .slick-dots {
  display: -webkit-box!important;
  display: -webkit-flex!important;
  display: -ms-flexbox!important;
  display: flex!important;
  position: absolute;
  left: 50%;
  -webkit-transform: translate(-50%);
  -moz-transform: translate(-50%);
  -ms-transform: translate(-50%);
  transform: translate(-50%);
  bottom: -32px;
}

.emotion-2 .slick-dots li {
  border-radius: 50%;
  height: 0.25rem;
  width: 0.25rem;
  margin-right: 0.25rem;
  opacity: 0.25;
}

.emotion-2 .slick-dots li:first-of-type {
  margin-left: 0.25rem;
}

.emotion-2 .slick-dots li button {
  display: none;
}

.emotion-2 .slick-dots li.slick-active {
  opacity: 1;
}

.emotion-2 .slick-dots li {
  background-color: #332E54;
}

.emotion-4 {
  width: 311px!important;
  min-height: 48px;
  margin: 0 0.5rem;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="emotion-4 emotion-5"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                Card 1
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
</div>
`;

exports[`<Carousel /> UI snapshots renders with props 1`] = `
.emotion-0 {
  max-width: 375px;
}

.emotion-2 {
  overflow: hidden!important;
  padding-bottom: 1.5rem;
}

.emotion-2 .slick-slider {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 1.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 .slick-list {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  overflow: hidden;
}

.emotion-2 .slick-list:focus {
  outline: none;
}

.emotion-2 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.emotion-2 .slick-slider .slick-track,
.emotion-2 .slick-slider .slick-list {
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  -o-transform: translate3d(0, 0, 0);
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
}

.emotion-2 .slick-track {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 .slick-loading .slick-track {
  visibility: hidden;
}

.emotion-2 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.emotion-2 .slick-slide.slick-loading img {
  display: none;
}

.emotion-2 .slick-slide.dragging img {
  pointer-events: none;
}

.emotion-2 .slick-initialized .slick-slide {
  display: block;
}

.emotion-2 .slick-loading .slick-slide {
  visibility: hidden;
}

.emotion-2 .slick-vertical .slick-slide {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  border: 1px solid transparent;
}

.emotion-2 .slick-arrow.slick-hidden {
  display: none;
}

.emotion-2 .slick-slider {
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 .slick-dots {
  display: -webkit-box!important;
  display: -webkit-flex!important;
  display: -ms-flexbox!important;
  display: flex!important;
  position: absolute;
  left: 50%;
  -webkit-transform: translate(-50%);
  -moz-transform: translate(-50%);
  -ms-transform: translate(-50%);
  transform: translate(-50%);
  bottom: -32px;
}

.emotion-2 .slick-dots li {
  border-radius: 50%;
  height: 0.25rem;
  width: 0.25rem;
  margin-right: 0.25rem;
  opacity: 0.25;
}

.emotion-2 .slick-dots li:first-of-type {
  margin-left: 0.25rem;
}

.emotion-2 .slick-dots li button {
  display: none;
}

.emotion-2 .slick-dots li.slick-active {
  opacity: 1;
}

.emotion-2 .slick-dots li {
  background-color: #332E54;
}

.emotion-4 {
  width: 311px!important;
  min-height: 48px;
  margin: 0 0.5rem;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="width: -100px; opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-center slick-current"
            data-index="0"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="emotion-4 emotion-5"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                Card 1
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
</div>
`;
